src/push-notifications/use-push-notifications.tsx

```diff
@@ -0,0 +1,131 @@
+import {useCallback, useState} from 'react';
+import {useRegister} from '@atb/push-notifications/use-register';
+import {useConfig} from '@atb/push-notifications/use-config';
+import {PermissionsAndroid, Platform} from 'react-native';
+import messaging, {
+  FirebaseMessagingTypes,
+} from '@react-native-firebase/messaging';
+import {usePreferences} from '@atb/preferences';
+
+type PermissionStatus = 'granted' | 'denied' | 'undetermined';
+
+export const usePushNotifications = () => {
+  const {
+    preferences: {language = 'nb'},
+  } = usePreferences();
+  const [permissionStatus, setPermissionStatus] =
+    useState<PermissionStatus>('undetermined');
+  const [isLoadingPermissionStatus, setIsLoadingPermissionStatus] =
+    useState<boolean>();
```

Comment:
`useState(false);` like on the line below is more clear imo

Reply:
I'm not sure if I get what you mean. If you are referring to the line break with `useState` on a new line, that's not something I can decide - it's defined (and enforced) by Prettier.

Reply:
```
const [isLoadingPermissionStatus, setIsLoadingPermissionStatus] =
    useState(false);
```

Reply:
Just instead of `useState<boolean>()`

Reply:
Resolved through introducing one common prop for status